commit 0cd6747277f1
Author: Valentin Gosu <valentin.gosu@gmail.com>
Date:   Tue Apr 10 22:07:47 2018 +0200

    Bug 1452417 - Hold a ref to mRequest in PACResolver::Notify. r=bagder, a=abillings
    
    MozReview-Commit-ID: 1QeFlAiTCNt
    
    --HG--
    extra : source : f7fb95c9979452fe25e42873f54cf56a80a0a0d8
---
 netwerk/base/ProxyAutoConfig.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git netwerk/base/ProxyAutoConfig.cpp netwerk/base/ProxyAutoConfig.cpp
index 2c965f8e8f4d..d562f321bb45 100644
--- netwerk/base/ProxyAutoConfig.cpp
+++ netwerk/base/ProxyAutoConfig.cpp
@@ -307,8 +307,9 @@ public:
   // nsITimerCallback
   NS_IMETHOD Notify(nsITimer *timer) override
   {
-    if (mRequest)
-      mRequest->Cancel(NS_ERROR_NET_TIMEOUT);
+    nsCOMPtr<nsICancelable> request(mRequest);
+    if (request)
+      request->Cancel(NS_ERROR_NET_TIMEOUT);
     mTimer = nullptr;
     return NS_OK;
   }
